sources/api/prairiedog/src/main.rs

```diff
@@ -253,8 +286,119 @@ fn load_crash_kernel() -> Result<()> {
     Ok(())
 }
 
+/// Queries Bottlerocket boot settings and generates initrd image file with boot config as the only data
+async fn generate_boot_config<P>(socket_path: P) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    let boot_settings = get_boot_config_settings(socket_path).await?;
+    info!("Generating initrd boot config from boot settings");
+    trace!("Boot settings: {:?}", boot_settings);
+    let bootconfig = to_string(&boot_settings).context(error::SerializeBootConfigSnafu)?;
+    trace!("Serializied boot config string: {}", bootconfig);
+    let initrd = generate_initrd(&bootconfig.into_bytes())?;
+    trace!("Writing initrd image file: {:?}", initrd);
+    tokio::fs::write(BOOTCONFIG_INITRD_PATH, &initrd)
+        .await
+        .context(error::WriteInitrdSnafu)?;
+    Ok(())
+}
+
+async fn get_boot_config_settings<P>(socket_path: P) -> Result<BootSettings>
+where
+    P: AsRef<Path>,
+{
+    let boot_settings = schnauzer::get_settings(socket_path)
+        .await
+        .context(error::RetrieveSettingsSnafu)?
+        .settings
+        .and_then(|s| s.boot)
+        .context(error::MissingSettingsSnafu)?;
+    Ok(boot_settings)
+}
+
+fn generate_initrd(bootconfig: &[u8]) -> Result<Vec<u8>> {
+    let mut bootconfig_size = bootconfig.len();
+    // Calculate checksum of the boot config file
```

Comment:
Nit: lesson learnt from my own PRs, explain the "why" instead of the "what", since the latter is obvious :+1: